### PR TITLE
[prometheus-operator] Upgrade to 44.2.0

### PR DESCRIPTION
**Pull request by QuentinBisson**

### CODE REVIEW

**Comment by QuentinBisson** [comment]
@giantswarm/team-atlas tested on WC and MC. Do you think this would require a new major version?

**Review by hervenicol — COMMENTED** [review]
I have a lot of stupid questions! :stuck_out_tongue_closed_eyes:

**Comment by hervenicol** [comment]
> @giantswarm/team-atlas tested on WC and MC. Do you think this would require a new major version?

Tough question.
For v2.0.0 that was a no-brainer because values format completely changed (nested chart).

Now:
* we didn't use to have a major version for each upstream version, even though these were major versions
* but we may want to change that. It will do a lot of major versions then!

File: helm/prometheus-operator-app/Chart.yaml
Comment (QuentinBisson) [reply helm/prometheus-operator-app/Chart.yaml]: Done

File: helm/prometheus-operator-app/README.md
Comment (QuentinBisson) [reply helm/prometheus-operator-app/README.md:3]: We should update it via CI but I do not have the time :( Can someone pick it up? Maybe @mcbenjemaa or @TheoBrigitte ?

File: helm/prometheus-operator-app/templates/_helpers.tpl
Comment (QuentinBisson) [reply helm/prometheus-operator-app/templates/_helpers.tpl:17]: No we get rid of the label hook and the network policy as they are added upstream

File: helm/prometheus-operator-app/values.yaml
Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml:60]: Yes it's a new component (disabled by default). You can now create a default ThanosRuler as you would a prometheus or an alertmanager from the values file. We just ensure we override the image to use ours (retagging)

File: helm/prometheus-operator-app/values.yaml
Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml:42]: We can add a note. Yes i got rid of it due to the use of VPA

Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml:42]: Note added in the changelog file

**Comment by QuentinBisson** [comment]
Well upstream is using major version 41 so we can probably do a major version bump. I'll you guys decide that next week

File: helm/prometheus-operator-app/README.md
Comment (mcbenjemaa) [reply helm/prometheus-operator-app/README.md:3]: I will take a look

File: helm/prometheus-operator-app/values.yaml
Comment (mcbenjemaa) [inline helm/prometheus-operator-app/values.yaml:49]: Hmmmm

File: helm/prometheus-operator-app/templates/_helpers.tpl
Comment (QuentinBisson) [reply helm/prometheus-operator-app/templates/_helpers.tpl:41]: No because the latest version upstream has it. I created a PR about it

File: helm/prometheus-operator-app/values.yaml
Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml:49]: Yes?

File: helm/prometheus-operator-app/values.yaml
Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml:47]: I'm not sure I understand this comment :D

File: changelog/32.x_41.x.md
Comment (QuentinBisson) [reply changelog/32.x_41.x.md:20]: Sure sure :)

**Comment by QuentinBisson** [comment]
@mcbenjemaa did you have time to work on the script?

**Comment by QuentinBisson** [comment]
@mcbenjemaa added to the board

File: helm/prometheus-operator-app/README.md
Comment (QuentinBisson) [reply helm/prometheus-operator-app/README.md]: I set it manually to 3.0.0 :)

File: helm/prometheus-operator-app/values.yaml
Comment (TheoBrigitte) [inline helm/prometheus-operator-app/values.yaml:46]: How does those images work in china? We need to be able to change the registry there.

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]

File: helm/prometheus-operator-app/values.yaml
Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml:46]: This is already set in 2.0.0

Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml:46]: Can we open an issue to add registry support upstream and create a new version of the app?

**Comment by QuentinBisson** [comment]
Tested on gremlin

**Comment by QuentinBisson** [comment]
@TheoBrigitte can I get another look?

File: helm/prometheus-operator-app/templates/_helpers.tpl
Comment (QuentinBisson) [inline helm/prometheus-operator-app/templates/_helpers.tpl:8]: @hervenicol this works with giantswarm/architect v4.26.0 in circle ci and validate enabled

**Comment by QuentinBisson** [comment]
@TheoBrigitte and @giantswarm/team-atlas can I get a review on this one? Once it's done I'll add the latest changes which include the global registry so it works in china :)

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM :)

File: changelog/32.x_41.x.md
Comment (mcbenjemaa) [inline changelog/32.x_41.x.md:8]: I just wanted to check this diff, but it's cumbersome :(

Comment (mcbenjemaa) [reply changelog/32.x_41.x.md:8]: I trust you 👍

File: helm/prometheus-operator-app/README.md
Comment (mcbenjemaa) [inline helm/prometheus-operator-app/README.md:21]: Why @hervenicol having Marie email :grin:

File: helm/prometheus-operator-app/values.yaml
Comment (mcbenjemaa) [inline helm/prometheus-operator-app/values.yaml:42]: I remember if this flag is omit, there's a condition about NetworkPolicy will fail!

File: CHANGELOG.md
Comment (mcbenjemaa) [inline CHANGELOG.md:10]: can you add a section to the root Readme as we did before: https://github.com/giantswarm/prometheus-operator-app#from-13x-to-2x

So, another section
```
### from-2.x to 3.x
```

File: helm/prometheus-operator-app/README.md
Comment (QuentinBisson) [reply helm/prometheus-operator-app/README.md:21]: Done

File: CHANGELOG.md
Comment (QuentinBisson) [reply CHANGELOG.md:10]: done

File: helm/prometheus-operator-app/values.yaml
Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml]: I upgraded to 42.0.2 so we can override the images now :)

Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml]: Not sure how it goes with WC in China though

**Comment by QuentinBisson** [comment]
I upgraded to 42.0.2 so I'll be testing it tomorrow

File: README.md
Comment (mcbenjemaa) [inline README.md:27]: please fix the link

File: README.md
Comment (mcbenjemaa) [inline README.md:29]: same here

**Comment by mcbenjemaa** [comment]
do we still need the cilium np? 
and why it's in the deployment-label-hook folder?

File: README.md
Comment (QuentinBisson) [reply README.md:29]: What is wrong with the links exactly?

Comment (mcbenjemaa) [reply README.md:29]: missing `blob/main`

```suggestion
* **⚠️ Please check the [breaking changes](https://github.com/giantswarm/prometheus-operator-app/blob/main/changelog/32.x_42.x.md) ⚠️**
```

**Comment by QuentinBisson** [comment]
Yes we need the cilium np because it's not upstream and it's in this webhook because it needs access to the api server

**Comment by QuentinBisson** [comment]
Blocked by https://github.com/prometheus-community/helm-charts/pull/2790 :(

**Comment by QuentinBisson** [comment]
@TheoBrigitte and @giantswarm/team-atlas do you mind taking another look? This was tested on gohan/med02 successfully :)

File: CHANGELOG.md
Comment (TheoBrigitte) [inline CHANGELOG.md:28]: We usually do not mention CI changes in changelog

File: CHANGELOG.md
Comment (TheoBrigitte) [inline CHANGELOG.md:24]: This section should be something else than `fixed`, probably `changed`.

File: CHANGELOG.md
Comment (TheoBrigitte) [inline CHANGELOG.md:40]: Why is this duplicated here ?

File: helm/prometheus-operator-app/README.md
Comment (TheoBrigitte) [inline helm/prometheus-operator-app/README.md:83]: Why was this removed ? It think its helpful

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (QuentinBisson) [reply CHANGELOG.md:28]: Removing. I just see there was a bit rebase issue in the changelog :(

File: CHANGELOG.md
Comment (QuentinBisson) [reply CHANGELOG.md:24]: Done

File: helm/prometheus-operator-app/README.md
Comment (QuentinBisson) [reply helm/prometheus-operator-app/README.md:83]: I agree but this is a duplicate that is also in the CONTRIBUTING.md file

File: CHANGELOG.md
Comment (QuentinBisson) [reply CHANGELOG.md]: Changelog is fixed, sorry about the mess. I did a git pull --rebase that went ok on the CLI but was a mess in files :(

**Comment by QuentinBisson** [comment]
@TheoBrigitte i need your unblocking :D

**Review by TheoBrigitte — APPROVED** [review]
LGTM when tested

**Comment by QuentinBisson** [comment]
Upgraded to 44.2.0 (with latest prometheus operator)

To test:
- [ ] Deployed as part of a bundle on MC
- [ ] Deployed as an app on a WC (works as default app)
- [ ] Test upgrade from 2.1.2

**Pull request merged**
